main/views.py

```diff
@@ -540,9 +626,9 @@ class DeleteView(LoginRequiredMixin, UserAllowedMixin, generic.DeleteView):
     """Generic delete view including login required and user allowed mixin and
     alternative for success message"""
 
-    def delete(self, request, *args, **kwargs):
+    def form_valid(self, form):
         messages.success(self.request, self.success_message)
-        return super(DeleteView, self).delete(request, *args, **kwargs)
+        return super(DeleteView, self).form_valid(form)
```

Comment:
Django 4 changed how deleteviews worked. For those wondering where this and other similar changes comes from.